api/test/models/location/location-data-mapper-impl.test.ts

```diff
@@ -0,0 +1,132 @@
+    // Configure Location Data Mapper
+    locationDataMapper = new LocationDataMapperImpl(acl, mysqlUow, new Logger());
+  });
+
+  afterEach(() => {
+    reset(mysqlUowMock);
+  });
+
+  describe('TEST: Location read', () => {
+    // @ts-ignore
+    it('generates the expected SQL to retrieve a location', async () => {
+      // GIVEN: An location with a valid ID to read from
+      const uid = 'test uid';
+      // WHEN: Retrieving data for this location
+      await locationDataMapper.get(uid);
+
+      // THEN: Generated SQL matches the expectation
+      const expectedSQL = 'SELECT * FROM `LOCATIONS` WHERE (uid= ?);';
+      const expectedParams = [uid];
+      const [generatedSQL, generatedParams] = capture<string, any[]>(mysqlUowMock.query)
+        .first();
+      verify(mysqlUowMock.query(anything(), anything(), anything())).once();
+      expect(generatedSQL).to.equal(expectedSQL);
+      expect(generatedParams).to.deep.equal(expectedParams);
+    });
+  });
+
+  describe('TEST: Location delete', () => {
+    // @ts-ignore
+    it('causes the location to get deleted', async () => {
+      // GIVEN: A location with a valid ID to read from
+      const uid = 'test uid';
+      // WHEN: Retrieving data for this location
+      await locationDataMapper.delete(uid);
+
+      // THEN: Generated SQL matches the expectation
+      const expectedSQL = 'DELETE FROM `LOCATIONS` WHERE (uid = ?);';
+      const expectedParams = [uid];
+      const [generatedSQL, generatedParams] = capture<string, any[]>(mysqlUowMock.query)
+        .first();
+      verify(mysqlUowMock.query(anything(), anything(), anything())).once();
+      expect(generatedSQL).to.equal(expectedSQL);
+      expect(generatedParams).to.deep.equal(expectedParams);
+    });
+  });
+
+  describe('TEST: Location count', () => {
+    // @ts-ignore
+    it('generates the expected SQL to retrieve the number of locations', async () => {
+      // GIVEN: Instance of a location data mapper
+      // WHEN: Retrieving number of locations
+      const result = await locationDataMapper.getCount();
+
+      // THEN: Generated SQL matches the expectation
+      const expectedSQL = 'SELECT COUNT(uid) AS "count" FROM `LOCATIONS`;';
+      const [generatedSQL] = capture<string>(mysqlUowMock.query).first();
+      verify(mysqlUowMock.query(anything(), anything(), anything())).once();
+      expect(generatedSQL).to.equal(expectedSQL);
+    });
+  });
+
+  describe('TEST: Location insert', () => {
+    // @ts-ignore
+    it('inserts the locations', async () => {
+      // GIVEN: A location to insert
+      const testLocation = new Location({
+        locationName: 'test location',
+      });
+      // WHEN: Retrieving number of locations
+      await locationDataMapper.insert(testLocation);
+
+      // THEN: Generated SQL matches the expectation
+      const expectedSQL = 'INSERT INTO `LOCATIONS` (`uid`, `location_name`) VALUES (?, ?);';
+      const expectedParams = [
+        testLocation.uid,

```

Comment:
This will be undefined, and the INSERT query should be `INSERT INTO `LOCATIONS` (`location_name`) VALUES (?);` since the did is auto generated by the database